File: java/反射/NO_Imort/Person.java
package a.b.c.d;
class Person {
  private String name;

  public void setName(String name) {
    this.name = name;
  }

  public String getName() {
    return this.name = name;
  }

  public Person() {
    System.out.println("Construct of Person");
  }

  public Person(String name) {
    this.name = name;
    System.out.println("Construct of Person, name is "+this.name);
  }
}

File: c++/base/person.cpp
#include "iostream"
#include "string"
#include "person.h"

using namespace std;
namespace A {
void Person::setName(string name)
{
  // name = n;
  this->name = name;
}
void Person::printfinfo(void)
{
  cout << "name = " << name << ", age = " << age << ", work = " << work << endl;
}
void Person::setWork(string work)
{
  this->work = work;
}

int Person::setAge(int age)
{
    if(age < 0 || age > 150) {
      this->age = 0;
      std::cout << "invalid age" << endl;
      return -1;
    } else {
      this->age = age;
      return 0;
    }
}

void printVersion()
{
  std::cout << "person_1.1" << endl;
}
}
